x-pack/elastic-agent/pkg/agent/operation/monitoring.go

```diff
@@ -324,6 +326,65 @@ func (o *Operator) getMonitoringMetricbeatConfig(output interface{}) (map[string
 			},
 		})
 	}
+
+	fixedAgentName := strings.ReplaceAll(agentName, "-", "_")
+	// setup cpu, memory and fd monitors for agents
+	modules = append(modules, map[string]interface{}{
+		"module":     "system",
+		"period":     "10s",
+		"metricsets": []string{"process"},
+		"index":      fmt.Sprintf("metrics-elastic_agent.%s-default", fixedAgentName),
+		"processes":  []string{install.BinaryName},
+		"processors": []map[string]interface{}{
+			{
+				"add_fields": map[string]interface{}{
+					"target": "data_stream",
+					"fields": map[string]interface{}{
+						"type":      "metrics",
+						"dataset":   fmt.Sprintf("elastic_agent.%s", fixedAgentName),
```

Comment:
Still question if it should have the ending `.elastic_agent`? Why not just `metrics-elastic_agent-default`? Or does that not match the logs?

Reply:
was thinking about following convention, i like shorter one better as well